src/ApacheModRewriteGenerator.php

```diff
@@ -63,20 +63,22 @@ public function generateRewrite( string $from, string $to, int $type ) : string
 			}
 		}
 
-		$output .= 'RewriteRule ^' . preg_quote(ltrim($fromPath, '/'), ' ') . '$ ' . $this->escapeSubstitution($prefix . ltrim($toPath, '/')) . '?' . $toQuery;
+		$output .= 'RewriteRule ^' . preg_quote(ltrim($fromPath, '/'), ' ') . '$ ' . $this->escapeSubstitution($prefix . ltrim($toPath, '/')) . $toQuery;
 
 		switch( $type ) {
 			case RewriteTypes::SERVER_REWRITE:
 				return "{$output}&%{QUERY_STRING}";
 			case RewriteTypes::PERMANENT_REDIRECT:
-				return "{$output} [L,R=301]";
+                        case RewriteTypes::TEMPORARY_REDIRECT:
+				return "{$output} [L,R=".RewriteTypes::name($type)."]";
 		}
 
 		throw new InvalidArgumentException("Unhandled RewriteType: {$type}", $type);
 	}
 
 	private function escapeSubstitution( string $input ) : string {
-		return preg_replace('/[-\s%$\\\\]/', '\\\\$0', $input);
+		//return preg_replace('/[-\s%$\\\\]/', '\\\\$0', $input);
+            return $input;
```

Comment:
I don't understand what you're up to here?

Escaping substitutions is important.

Reply:
My experiences with your tool show me that this is not necessary to escape the rewrited URL.

Reply:
`foo.html bar%20baz.html` is an example of where it is necessary. 

As you have it, it outputs `RewriteRule ^foo\.html$ /bar baz.html&%{QUERY_STRING}` which results in an internal server error. The space needs to be escaped.